src/varname.jl

```diff
@@ -89,7 +92,12 @@ getindexing(vn::VarName) = vn.indexing
 Base.hash(vn::VarName, h::UInt) = hash((getsym(vn), getindexing(vn)), h)
 Base.:(==)(x::VarName, y::VarName) = getsym(x) == getsym(y) && getindexing(x) == getindexing(y)
```

Comment:
We need to be careful here now... what about `DynamicIndexLens`, containing a closure?

Reply:
In that case they won't be equal, right? Which IMO seems like sensible behavior, no? They're not the same but they have the same behavior. 

Seems like the only other alternative is to completely drop the definition of `==` which also seems a bit much?

Reply:
Agree to both, this was more like... warn the _callers_ of that, and check that we ourselves don't fall over this (which should be easy, though, as we are only concerned with concretized varnames anyway).

Reply:
Sure :+1: How should do this though? Warning will be slow. Error is too much. Docstring?